### ISSUE TITLE
The Mui X Data Grid does not resize correctly within the Toolpad core dashboard layout.

### ISSUE THREAD

**Issue by OrwaPog** [mui/mui-x#15901]
### Search keywords

Mui X data grid toolpad core

### Latest version

- [X] I have tested the latest version

### Affected products

Data Grid

### Steps to reproduce

Steps:
1. Setup toolpad core dashboard layout
2.create a random data grid and put it inside the layout
3.closing and opening the drawer should reproduce the issue

### Current behavior

https://github.com/user-attachments/assets/1f38ca7d-45fa-45ad-9db8-307abd33dc17

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by oliviertassinari** [comment]
It's a Toolpad Core bug, no? Same as https://github.com/mui/mui-x/issues/11295#issuecomment-2232925715?

**Comment by michelengelen** [comment]
I agree @oliviertassinari ... I'll transfer and close this as a duplicate. 👍🏼
